### DISCUSSION TITLE
How to build human in the loop experience in langserve.

### DISCUSSION THREAD

**Discussion by SoulEvill** [langchain-ai/langserve#654] [General]
I am building an agent that is based on the langgraph, when it goes through certain state it will pop up for user input.

I would love to serve this through langserve - wondering if langserve would able to support this use case? or general idea how to build it out would be really appreaciated.

Thanks!

**Comment by Aillian** [comment]
have you figured this one out?

**Reply by mohith-jagalmohan** [reply]
I am also trying to figure this out

**Reply by eyurtsev** [reply]
Use langgraph platform with interrupt(). https://langchain-ai.github.io/langgraph/concepts/human_in_the_loop/

**Reply by Aillian** [reply]
The implementation of `interrupt` is very clear, one thing i couldn't figure out is how to pass `Command(resume=...)` in invoke with Langserve? @eyurtsev

**Reply by aamaragones** [reply]
@Aillian and @mohith-jagalmohan... I'm encountering, I just need to send but give me Entity Error...

`graph.stream(Command(resume={"action": "continue"}), config, stream_mode="updates"):`

Its lookslike we can reproduce the __interrupt__ but we cant send the Command due deprecation of Langserver...

https://github.com/langchain-ai/langserve/discussions/790 

Here is the Migration guide...

https://github.com/langchain-ai/langserve/blob/main/MIGRATION.md

**Reply by Aillian** [reply]
@EiinGeeeL Twisting our arms to use their paid solution and depreciating an open source one is really frustrating!

If anyone knows any alternative open source solutions or ways to use features like human in the loop please share them...
